### PR TITLE
fw pos ctrl: fix state switching logic for takeoff and landing

### PR DESCRIPTION
**Pull request by tstastny**
**Describe problem solved by this pull request**
Takeoff and landing modes were separated from the `control_auto()` umbrella in #19495 in an effort to start untangling the fixed-wing position control module before some more significant refactoring to come. However, some of the internal state resets  and position setpoint handling logic was not exactly translated in a way that reproduced the previous behavior. Sorry - I missed them as well in the PR review.

Example: takeoff mode would be overridden from this logic:
https://github.com/PX4/PX4-Autopilot/blob/14a2fdfe551435638e22864437e75030abbf28eb/src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp#L760-L766

And some TECS states and other attitude integral resets were not initialized in takeoff and landing modes as they were before. 

**Describe your solution**
This PR is a hotfix to revive the same behavior the module had prior to #19495, while keeping the takeoff and landing modes separated. There are more fundamental issues that need work in this module.. e.g. `in_takeoff_situation()` is a quite egregious one IMHO. But these can be saved for a subsequent discussion and PR. 

**Test data / coverage**
Tested that the basic functionality operates as it did before. E.g. making sure the takeoff mode is not exited until navigator sees the mission item reached, loiter down to waypoints work (the position setpoint switching), flying back to takeoff waypoints later in a mission does not re-enter takeoff logic, etc.

https://user-images.githubusercontent.com/8026163/166902577-a16f9fea-6010-46be-a9c2-f1b3434d9a48.mp4

Note the takeoff behavior is actually kind of terrible (no loiter up to takeoff point when below the takeoff point altitude) - but this has been the behavior for some time now. I'm working on a larger refactor of takeoff and landing for fixed-wing which will simplify these modes and also allow for a more general discussion on what takeoff actually *should* look like. But for now.. this PR is just to make sure the next release has the behavior that's been expected.

@sfuhrer would be great if you could check that I did not mess up any VTOL functionality.
@dagar if the current release branch has #19495 in it .. we would probably need to include this PR as well to make sure the logic isnt broken on the release.

### CODE REVIEW

**Review by Jaeyoung-Lim — DISMISSED** [review]
Thanks for the fix!

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 1]
```diff
@@ -909,6 +918,20 @@ FixedwingPositionControl::control_auto(const hrt_abstime &now, const Vector2d &c
 		publishOrbitStatus(current_sp);
 	}
 
+	// update lateral guidance timesteps for slewrates
+	if (_param_fw_use_npfg.get()) {
+		_npfg.setDt(dt);
```
Comment (sfuhrer) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:923]: Not an issue here, but isn't the setting of Dt missing still for control_auto_descend(), control_auto_fixed_bank_alt_hold() and control_manual_position()? Shouldn't we just do it outside, in the main Run() function? Then we also just call update_position_control_mode_timestep() right? Or am I missing something?

**Comment by sfuhrer** [comment]
@tstastny thanks a lot for taking care of these things - I tested VTOL in SITL without finding a regression. Many of the points you fixed are though likely not easy to spot in SITL (things like the takeoff help, integrator lockings and dt updates).

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp (hunk 1 above)
Comment (tstastny) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:923]: good catch - `control_manual_position()` should also have the dt update for npfg and l1. `descend` and `fixed_bank_alt_hold` dont use the guidance laws, so no need there. 

@dagar you were the one who originally moved the dt calculation into each resp. control method. if we are comparing these all to the `_local_pos.timestamp` anyway, there shouldn't be any issue to update the slewrates in `run()`, right?

Comment (tstastny) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:923]: FYI - centralized the dt calculation in a new PR #19595

Comment (sfuhrer) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:923]: alright :+1: 
Do you still want to add it to control_manual_position() already for this PR?

**Comment by tstastny** [comment]
>Do you still want to add it to control_manual_position() already for this PR?
@sfuhrer ok added. -- as for the cleanup (centralizing the control interval calculation) will leave that to #19595, and we'll keep this one a hot fix just to provide the functionality that was existing for this release.

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 2]
```diff
@@ -1512,7 +1512,7 @@ FixedwingPositionControl::control_auto_takeoff(const hrt_abstime &now, const Vec
 		if (_param_fw_use_npfg.get()) {
 			_npfg.setAirspeedNom(target_airspeed * _eas2tas);
 			_npfg.setAirspeedMax(_param_fw_airspd_max.get() * _eas2tas);
-			_npfg.navigateWaypoints(prev_wp_local, curr_wp_local, curr_pos_local, ground_speed, _wind_vel);
+			_npfg.navigateWaypoints(curr_wp_local, curr_wp_local, curr_pos_local, ground_speed, _wind_vel);
```
Comment (sfuhrer) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:1515]: could you maybe add a small code comment here on why this change was necessary - like // pass curr_wp_local instead of prev_wp_local to disable line tracking and instead make it track the takeoff WP as a single point

Comment (tstastny) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:1517]: was already on it :) -- see recent force push
